File: java/ListAllProjectsData.java
//
//	GitHub: https://github.com/SERPWoo/API
//
//	This code requests all of your projects and outputs their ID, names, amount of keywords, and links to API query of keywords
//
//	This output is text format
//
//	Last updated - Aug 30th, 2017 @ 10:40 EST (@MercenaryCarter https://github.com/MercenaryCarter and https://twitter.com/MercenaryCarter)
//
//	Compile Command: javac ListAllProjectsData.java
//	Run Command: java ListAllProjectsData
//

import java.net.*;
import java.io.*;
import org.json.*;
 
public class ListAllProjectsData
{
  public static void main(String[] args)
  {
	// Get your API Key here: https://www.serpwoo.com/v3/api/ (should be logged in)
  	String API_key = "API_KEY_HERE";

    String output  = getUrlContents("https://api.serpwoo.com/v1/projects/?key=" + API_key);
	
	
	JSONObject json = new JSONObject(output); // Convert text to object
	System.out.println(json.toString(4));
	
    //System.out.println(output);
  }

  private static String getUrlContents(String theUrl)
  {
    StringBuilder content = new StringBuilder();

    // many of these calls can throw exceptions, so i've just
    // wrapped them all in one try/catch statement.
    try
    {
      // create a url object
      URL url = new URL(theUrl);

      // create a urlconnection object
      URLConnection urlConnection = url.openConnection();

      // wrap the urlconnection in a bufferedreader
      BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(urlConnection.getInputStream()));

      String line;

      // read from the urlconnection via the bufferedreader
      while ((line = bufferedReader.readLine()) != null)
      {
        content.append(line + "\n");
      }
      bufferedReader.close();
    }
    catch(Exception e)
    {
      e.printStackTrace();
    }
    return content.toString();
  }
}

File: go/list-all-projects-data.go
//
// Github Repo: https://github.com/SERPWoo/API
//
// This code requests all of your projects' data in JSON format
//
// Last updated - Aug 26th, 2017 @ 16:35 EST (@MercenaryCarter https://github.com/MercenaryCarter and https://twitter.com/MercenaryCarter)
//
// Run Command: go run list-all-projects-data.go
//

	package main

		import (
			"fmt"
			"io/ioutil"
			"net/http"
		)

func main() {
	var client http.Client

		// Get your API Key here: https://www.serpwoo.com/q/api/ (should be logged in)
		var API_key = "API_KEY_HERE"

	resp, err := client.Get("https://api.serpwoo.com/v1/projects/?key=" + API_key)

		if err != nil {
			// There is a problem
		}

	defer resp.Body.Close()

		if resp.StatusCode == 200 { // OK
			bodyBytes, err2 := ioutil.ReadAll(resp.Body)

			if err2 == nil {
				bodyString := string(bodyBytes)
				fmt.Println(bodyString)				
			}

		}

}